File: src/main/java/io/github/bot/terminal/application/domains/notificarion_api/rest/dto/NotificationApiDTO.java
package io.github.bot.terminal.application.domains.notificarion_api.rest.dto;

import lombok.Data;

@Data
public abstract class NotificationApiDTO {
    public NotificationApiDTO(String type) {
        this.type = type;
    }

    protected String id;
    protected String label;
    protected String state;
    private String type;
}

File: src/main/java/io/github/bot/terminal/application/domains/notificarion_api/rest/dto/NotificationApiDTO.kt
package io.github.bot.terminal.application.domains.notificarion_api.rest.dto

import io.github.bot.terminal.application.domains.notificarion_api.values.NotificationApiType

abstract class NotificationApiDTO(val type: String)

data class SlackNotificationApiDTO(
        val id: String,
        val label: String,
        val enabled: Boolean,
        val token: String,
        val chanel: String
) : NotificationApiDTO(NotificationApiType.Constants.SLACK_BOT)

data class TelegramNotificationApiDTO(
        val id: String,
        val label: String,
        val enabled: Boolean,
        val token: String,
        val botFatherName: String
) : NotificationApiDTO(NotificationApiType.Constants.TELEGRAM_BOT)